nixos/modules/config/sysctl.nix

```diff
@@ -21,19 +21,27 @@ in
   options = {
 
     boot.kernel.sysctl = mkOption {
-      type = types.submodule {
+      type = let
+        highestValueType = types.nullOr types.ints.unsigned // {
```

Comment:
It would be good to put highestValueType to lib/types.nix.  Not a blocker for this PR though.